Author: Luca Boccassi <bluca@debian.org>
Description: add preprocessor definitions for mips64el
Forwarded: yes
Last-Update: 2017-10-18
--- a/src/prelude.h
+++ b/src/prelude.h
@@ -91,7 +91,8 @@
 
 #if (defined (__64BIT__) || defined (__x86_64__) || defined (__AARCH64EL__) \
     || defined (__PPC64__) || defined (__powerpc64__) || defined (__ppc64__) \
-    || defined (__s390x__) || (defined (__sparc__) && defined (__arch64__)))
+    || defined (__s390x__) || (defined (__sparc__) && defined (__arch64__)) \
+    || (defined (__mips__) && defined (_MIPSEL) && _MIPS_SIM == _ABI64))
 #    define __IS_64BIT__                /*  May have 64-bit OS/compiler      */
 #else
 #    define __IS_32BIT__                /*  Else assume 32-bit OS/compiler   */
--- a/src/sfl.h
+++ b/src/sfl.h
@@ -130,7 +130,8 @@
 
 #if (defined (__64BIT__) || defined (__x86_64__) || defined (__AARCH64EL__) \
     || defined (__PPC64__) || defined (__powerpc64__) || defined (__ppc64__) \
-    || defined (__s390x__) || (defined (__sparc__) && defined (__arch64__)))
+    || defined (__s390x__) || (defined (__sparc__) && defined (__arch64__)) \
+    || (defined (__mips__) && defined (_MIPSEL) && _MIPS_SIM == _ABI64))
 #    define __IS_64BIT__                /*  May have 64-bit OS/compiler      */
 #else
 #    define __IS_32BIT__                /*  Else assume 32-bit OS/compiler   */
